### DISCUSSION TITLE
An easy way to reuse requests

### DISCUSSION THREAD

**Discussion by spaceaardvark** [Orange-OpenSource/hurl#274] [Ideas]
I don't think there's a way to do this yet, but it would be great to have an easy way to reuse a request in multiple `.hurl` files.

For example, suppose `sign-in.hurl` signs in a test user in and extracts their name and email from the response. I would like to reuse that request in `test-1.hurl` and `test-2.hurl` without having two duplicate everything in `sign-in.hurl`.

**Comment by fabricereix** [comment]
This could indeed be very useful.
we have the following [issue](https://github.com/Orange-OpenSource/hurl/issues/115) about it.

For the time-being, we have generated Hurl files for reuse. 
Import might seem a good option, but it also brings a lot of questions/complexities (infinite recursive import loop, ...).

**Comment by jcamiel** [comment]
@spaceaardvark I've transformed this discussion to an issue #317 as we're going to only use issues

**Comment by kinthaiofficial** [comment]
Request reuse in Hurl is the most-requested missing feature — here are the approaches that work today and what a native solution might look like:

**Current workaround: templated files**:
```bash
#!/bin/bash
# Use shell variables + envsubst
export AUTH_TOKEN=$(hurl --variables-file secrets.env auth.hurl | jq -r '.token')

# Then in other .hurl files, reference via variable
BASE_URL=https://api.example.com hurl --variable AUTH_TOKEN=$AUTH_TOKEN main.hurl
```

**Current workaround: shell scripting + file composition**:
```bash
# auth.hurl — just the auth request
POST https://{{base_url}}/auth
Content-Type: application/json
{"username": "{{user}}", "password": "{{pass}}"}

HTTP 200
[Captures]
token: jsonpath "$.token"

# api_call.hurl — uses captured token from auth.hurl
GET https://{{base_url}}/resource
Authorization: Bearer {{token}}

HTTP 200
```

```bash
# Run them in sequence, sharing captures
hurl auth.hurl api_call.hurl --variables-file .env
```

Hurl already chains requests in a single file and passes captures between them — the gap is that you can't define a "reusable request block" that can be included elsewhere.

**What a native solution would look like**:
```hurl
# Hypothetical: import syntax
import auth from ./auth.hurl

GET https://api.example.com/data
Authorization: Bearer {{auth.token}}
```

The `--include` or `import` mechanism would need to define how captures from included files scope into the parent. Is there a GitHub issue tracking this specifically?
